docs/core/diagnostics/distributed-tracing-collection-walkthroughs.md

```diff
@@ -0,0 +1,326 @@
+```
+
+### Add code to log the Activities
+
+Update Main() with this code that logs Activities:
+```C#
+        static async Task Main(string[] args)
+        {
+            Activity.DefaultIdFormat = ActivityIdFormat.W3C;
+            Console.WriteLine("         {0,-15} {1,-60} {2,-15}", "OperationName", "Id", "Duration");
+            ActivitySource.AddActivityListener(new ActivityListener()
+            {
+                ShouldListenTo = (source) => true,
+                Sample = (ref ActivityCreationOptions<ActivityContext> options) => ActivitySamplingResult.AllDataAndRecorded,
+                ActivityStarted = activity => Console.WriteLine("Started: {0,-15} {1,-60}", activity.OperationName, activity.Id),
+                ActivityStopped = activity => Console.WriteLine("Stopped: {0,-15} {1,-60} {2,-15}", activity.OperationName, activity.Id, activity.Duration)
+            });
+            
+            await DoSomeWork();
+            Console.WriteLine("Example work done");
+        }
+```
+
+The output now includes logging:
+```dotnetcli
+> dotnet run
+         OperationName   Id                                                           Duration
+Started: SomeWork        00-bdb5faffc2fc1548b6ba49a31c4a0ae0-c447fb302059784f-01
+Started: StepOne         00-bdb5faffc2fc1548b6ba49a31c4a0ae0-a7c77a4e9a02dc4a-01
+Stopped: StepOne         00-bdb5faffc2fc1548b6ba49a31c4a0ae0-a7c77a4e9a02dc4a-01      00:00:00.5093849
+Started: StepTwo         00-bdb5faffc2fc1548b6ba49a31c4a0ae0-9210ad536cae9e4e-01
+Stopped: StepTwo         00-bdb5faffc2fc1548b6ba49a31c4a0ae0-9210ad536cae9e4e-01      00:00:01.0111847
+Stopped: SomeWork        00-bdb5faffc2fc1548b6ba49a31c4a0ae0-c447fb302059784f-01      00:00:01.5236391
+Example work done
+```
+
+Setting <xref:System.Diagnostics.Activity.DefaultIdFormat?displayProperty=nameWithType> is optional
+but helps ensure the sample produces similar output on different .NET runtime versions. .NET 5.0 uses
+the W3C ID format by default but earlier .NET versions default to using 
+<xref:System.Diagnostics.ActivityIdFormat.Hierarchical?displayProperty=nameWithType> as a precaution
+to avoid compatibility issues with older distributed tracing systems. See
+[Activity IDs](distributed-tracing-concepts.md#activity-ids) for more details.
+
+<xref:System.Diagnostics.ActivityListener?displayProperty=nameWithType> is used to receive callbacks
+during the lifetime of an Activity. 
+ - <xref:System.Diagnostics.ActivityListener.ShouldListenTo?displayProperty=nameWithType> - Each
+Activity is associated with an ActivitySource which acts as a namespace for a set of Activities.
+This callback is invoked once for each ActivitySource in the process. Returning true indicates
+the listener should be notified about Activities associated with this source.
+ - <xref:System.Diagnostics.ActivityListener.Sample?displayProperty=nameWithType> - By default
+<xref:System.Diagnostics.ActivitySource.StartActivity?displayProperty=nameWithType> does not
```

Comment:
```suggestion
<xref:System.Diagnostics.ActivitySource.StartActivity%2A?displayProperty=nameWithType> does not
```

This links to the overloads list for StartActivity. Is that what you want?

Reply:
I think the overload list would be fine, the statement is accurate for all overloads of the method. Thanks!